src/figuresClasses.ts

```diff
@@ -1,11 +1,91 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function checkLengths(errorMessage: string, ...lengths: number[]): void {
+  if (lengths.some((length) => length <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
The constructor parameters are being manually assigned to class properties which is redundant when using TypeScript's parameter properties. You can remove the assignments and declare the properties with access modifiers directly in the constructor signature.